app/models.py

```diff
@@ -2551,3 +2551,90 @@ def serialize(self):
             "original_filename": self.original_filename,
             "page_count": self.page_count,
         }
+
+
+class UnsubscribeRequestReport(db.Model):
+    __tablename__ = "unsubscribe_request_report"
+    """
+    CREATE VIEW unsubscribe_request_report AS
+    SELECT
+        service_id,
+        unsubscribe_request_report_id as id,
+        min(created_at) as earliest_timestamp,
+        max(created_at) as latest_timestamp,
+        (case when downloaded_at is null then false else true end) as has_been_downloaded
+        count(*)
+    FROM unsubscribe_requests
+    group by service_id, unsubscribe_request_report_id
+    """
+
+    # Tell alembic not to create this as a table. We have a migration where we manually set this up as a view.
+    # This is custom logic we apply - not built-in logic. See `migrations/env.py`
+    __table_args__ = {"info": {"managed_by_alembic": False}}
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True)
+    service_id = db.Column(UUID(as_uuid=True), db.ForeignKey("services.id"), nullable=False)
+    service = db.relationship(Service, backref=db.backref("unsubscribe_requests"))
+
+    earliest_timestamp = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
+    latest_timestamp = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
+    has_been_downloaded = db.Column(db.Boolean, nullable=False)
+    count = db.Column(db.Integer, nullable=False)
+
+
+class UnsubscribeRequest(db.Model):
+    __tablename__ = "unsubscribe_request"
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+
+    # WIP:
+    # Ignoring a strict foreign key relationship here for now. Notifications are archived to the NotificationHistory
+    # table by a nightly job and I haven't investigated whether that might break a strict FK yet or if it would
+    # work smoothly. We can still have a relationship using an explicit join condition.
+    notification_id = db.Column(UUID(as_uuid=True), index=True, nullable=False)
+    notification = db.relationship(
+        "NotificationAllTimeView",
+        primaryjoin="NotificationLetterDespatch.notification_id == foreign(NotificationAllTimeView.id)",
```

Comment:
Should this be `primaryjoin="UnsubscribeRequest.notification_id == foreign(NotificationAllTimeView.id)"`?

Reply:
oops, typo. squashed out